src/main/scala/Api.scala

```diff
@@ -82,14 +85,34 @@ class Api(kafkaClientActorRef: ActorRef)
 
   def healthCheck: StandardRoute = {
     getHealth match {
-      case Health(true, message, None) => complete(Json.toJson(message).toString())
+      case Health(true, message, None) => complete(message)
       case Health(false, _, Some(e)) => failWith(e)
     }
   }
 
   def getHealth: Health = {
     val health = healthCheck("Health") {
-      Result.healthy("OK")
+
+      val checkDuration = timeoutDuration.duration
+      val healthFuture = askFor[DescribeClusterResult](DescribeKafkaCluster).map { clusterDesc =>
+          val clusterNodes = clusterDesc.nodes.get(checkDuration.length, checkDuration.unit)
+          val clusterId = clusterDesc.clusterId.get(checkDuration.length, checkDuration.unit)
+          logger.debug(s"clusterId: $clusterId; nodes: $clusterNodes")
+
+          val clusterIdAvailable: Boolean = Option(clusterId).isDefined
+          if (clusterIdAvailable && !clusterNodes.isEmpty) {
+            Result.healthy("OK")
+          } else {
+            Result.unhealthy("Error connecting to Kafka Cluster")
+          }
+        }
+
+      try {
+        Await.result[Result](healthFuture, checkDuration)
```

Comment:
never use Await in scala. We can let AKKA HTTP handle this with [completeOrRecoverWith](https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/future-directives/completeOrRecoverWith.html) inside the def healthcheck (this def will look a little redundant after this so could also be refactored out). Just replace the complete(message) with completeOrRecoverwith(getHealth) and remove the match

Reply:
Sorry, first time using Akka. 

Tried this, it didn't like it 

```scala
  def healthCheck: StandardRoute = {
    completeOrRecoverWith(getHealth) { e =>
      failWith(e)
    }
  }

  def getHealth: Future[Result] = {
      val checkDuration = timeoutDuration.duration
      askFor[DescribeClusterResult](DescribeKafkaCluster).map { clusterDesc =>
          val clusterNodes = clusterDesc.nodes.get(checkDuration.length, checkDuration.unit)
          val clusterId = clusterDesc.clusterId.get(checkDuration.length, checkDuration.unit)
          logger.debug(s"clusterId: $clusterId; nodes: $clusterNodes")

          val clusterIdAvailable: Boolean = Option(clusterId).isDefined
          if (clusterIdAvailable && !clusterNodes.isEmpty) {
            Result.healthy("OK")
          } else {
            Result.unhealthy("Error connecting to Kafka Cluster")
          }
        }
  }
```

says `Type mismatch, expected CompleteOrRecover, actual: Future[HealthCheck.Result]` on `getHealth`... which I find strange because I can do `Future { "OK" }` in there instead, like shown in the docs, and it'll accept that Future just fine

Reply:
if you look at the diff theres

`def getHealth()
    val health = healthCheck("Health") {
      ....
     }.execute()
`
It's been a while since i wrote that, but think it was something to do with rendering strings.